### PR TITLE
Refactor header rails_admin

### PR DESCRIPTION
**Pull request by lannp**

### CODE REVIEW

**Comment by hienbx94** [comment]
Ok

**Comment by ttuan** [comment]
OK

**Comment by VuongLong** [comment]
ok

**Comment by minhld1709** [comment]
OK

File: app/controllers/rails_admin/main_controller.rb [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+      return nil unless params[:associated_collection].present?
+      source_abstract_model = RailsAdmin::AbstractModel.new(to_model_name(params[:source_abstract_model]))
+      source_model_config = source_abstract_model.config
+      source_object = source_abstract_model.get(params[:source_object_id])
+      action = params[:current_action].in?(%w(create update)) ? params[:current_action] : 'edit'
+      @association = source_model_config.send(action).fields.detect { |f| f.name == params[:associated_collection].to_sym }.with(controller: self, object: source_object)
+      @association.associated_collection_scope
+    end
+  end
+end
```
Comment (hellovietnam93) [inline app/controllers/rails_admin/main_controller.rb]: convention

**Comment by hellovietnam93** [comment]
OK

File: app/models/concerns/rails_admin_subject.rb [hunk 2]
```diff
@@ -4,8 +4,17 @@ module RailsAdminSubject
   included do
     rails_admin do
       list do
-        field :name
+        filters [:courses]
+        field :name do
+         formatted_value do
+            bindings[:view].content_tag(:a, "#{bindings[:object].name}" ,
```
Comment (hellovietnam93) [inline app/models/concerns/rails_admin_subject.rb]: thừa dấu cách

**Comment by vietanhbkaaa** [comment]
Ok

**Pull request merged**